### ISSUE TITLE
yarn build:google raises error

### ISSUE THREAD

**Issue by ceritium** [DecliningLotus/fontsource#507]
**Describe the bug**
`yarn build:google` raises an error

```bash
...
Finished processing alef
Downloading alegreya-sans-sc
/Users/josegalisteo/code/fontsource/node_modules/p-cancelable/index.js:48
					throw new Error('The `onCancel` handler was attached after the promise settled.');
           ^
Error: The `onCancel` handler was attached after the promise settled.
    at onCancel (/Users/josegalisteo/code/fontsource/node_modules/p-cancelable/index.js:48:12)
    at makeRequest (/Users/josegalisteo/code/fontsource/node_modules/got/dist/source/as-promise/index.js:38:13)
    at Request.<anonymous> (/Users/josegalisteo/code/fontsource/node_modules/got/dist/source/as-promise/index.js:143:17)
    at Object.onceWrapper (node:events:510:26)
    at Request.emit (node:events:390:28)
    at Request.emit (node:domain:475:12)
    at Timeout.retry (/Users/josegalisteo/code/fontsource/node_modules/got/dist/source/core/index.js:1278:30)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
error Command failed with exit code 1.
...
```

**To Reproduce**

1. Clone the repo
2. yarn install
3. yarn build:google 

**Expected behavior**
To not raise an error

**Additional context**

Branch: main
commit: ab1ac9d30b675af005f4946a75ad43e3f3bb2d6d
node: v16.13.0
yarn: 1.22.17

**Comment by ayuhito** [comment]
This is the awkward race condition we sometimes face using Got. Sometimes running it again fixes it? I've managed to tweak the settings so it should rarely happen with Github Actions (which is really the only time that command should be used), but it could break on other machines or environments. 

If you change [this line](https://github.com/fontsource/fontsource/blob/main/scripts/google/download-google.ts#L32) from 3 to 1, you shouldn't experience the error. But your build time will take a long long time (anywhere from half an hour to couple hours).

I've tried switching off Got in the past but then build times rise exponentially so it's a very finicky and hard to test at scale thing to change. But  I'm planning to update this project to ESM once I switch out Jest with Vitest. I can then try updating Got and see if that helps things out.
